app/services/invoice_pdf_generator.rb

```diff
@@ -101,9 +101,15 @@ def add_client_info
     @composer.text(@input[:client_address], margin: margin_bottom(3))
   end
 
-  # rubocop:disable Metrics/MethodLength
   def add_items_table
-    header = lambda do |_tb|
+    data = @input[:items].each_with_index.map { |item, i| build_item_row(item, i + 1) }
+    data << build_total_row
+
+    @composer.table(data, column_widths: [-1, -9, -3, -2, -2, -3], header: items_table_header, margin: margin_bottom(2))
+  end
+
+  def items_table_header
+    lambda do |_tb|
```

Comment:
I don't know if I prefer to keep the `lambda` here to have a method that returns a callable as expected by the `@composer.table(header:)` argument, or to replace it with the plain array and pass a reference to the method to `.table(header:)`:

```suggestion
    @composer.table(data, column_widths: [-1, -9, -3, -2, -2, -3], header: method(:items_table_header), margin: margin_bottom(2))
  end

  def items_table_header
    # remove the lambda do and the "end"
```